File: Programming Skills II/150. Evaluate Reverse Polish Notation/150. Evaluate Reverse Polish Notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        int t, t2;
        for(int i=0; i<tokens.size(); ++i){
            if(tokens[i] == "+"){
                t = st.top();
                st.pop();
                t2 = st.top();
                st.pop();
                st.push(t+t2);
            }else if(tokens[i] =="-"){
                t = st.top();
                st.pop();
                t2 = st.top();
                st.pop();
                st.push(t2-t);
            }else if(tokens[i] =="*"){
                t = st.top();
                st.pop();
                t2 = st.top();
                st.pop();
                st.push(t2*t);
            }else if(tokens[i] =="/"){
                t = st.top();
                st.pop();
                t2 = st.top();
                st.pop();
                st.push(t2/t);
            }else{
                st.push(stoi(tokens[i]));
            }
        }
        return st.top();
    }
};

File: Programming Skills II/150. Evaluate Reverse Polish Notation/150. Evaluate Reverse Polish Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for i in tokens:
            if i == "+":
                t = stack.pop()
                t1 = stack.pop()
                stack.append(t+t1)
            elif i == "-":
                t = stack.pop()
                t1 = stack.pop()
                stack.append(t1-t)
            elif i == "*":
                t = stack.pop()
                t1 = stack.pop()
                stack.append(t1*t)
            elif i == "/":
                t = stack.pop()
                t1 = stack.pop()
                stack.append( int(t1/t) )
            else:
                stack.append(int(i))
        
        return stack.pop()
            
